prompt_toolkit/input/win32.py

```diff
@@ -315,6 +318,37 @@ def _get_keys(
                     for key_press in self._handle_mouse(ev):
                         yield key_press
 
+    @staticmethod
+    def _merge_paired_surrogates(key_presses: List[KeyPress]) -> Iterator[KeyPress]:
```

Comment:
Just wondering, is this similar to what ``unicodedata.normalize`` does, but for ``KeyPress`` objects?

Reply:
Not exactly, although a similar idea. unicodedata.normalize works at a higher level of abstraction, between "conceptual" characters and code points, i.e. it can merge 'a' and '◌̈' into 'ä'. This function is merging UTF-16 surrogate pair code units (see https://en.wikipedia.org/wiki/UTF-16) into single code points. Apparently the underlying windows API calls are working in UTF-16 and treating the code units as individual characters, when it ideally should return full code points, hence this workaround.

Reply:
Example: the character 😀 (U+1F600) can be encoded with the two 16-bit code units [0xD83D, 0xDE00].